src/components/map/toolbar/EventLog.tsx

```diff
@@ -14,16 +19,86 @@ interface IEventLogProps {
 
 @observer
 export default class EventLog extends React.Component<IEventLogProps, IEventLogState> {
+    scrollRef: React.RefObject<HTMLDivElement>;
+
     constructor(props: IEventLogProps) {
         super(props);
         this.state = {
             isOpen: false,
             isAlertShown: false,
         };
+        this.scrollRef = React.createRef();
+    }
+
+    private getLogRow = (action: logActions | null, objectId: string, entity: entityNames) => {
+        let logActionClass = '';
+        switch (action) {
+        case logActions.ADD:
+            logActionClass = s.logActionAdd;
+            break;
+        case logActions.DELETE:
+            logActionClass = s.logActionDelete;
+            break;
+        case logActions.MOVE:
+            logActionClass = s.logActionMove;
+            break;
+        }
+
+        return (
+            <div className={s.logRow}>
+                <div className={classnames(s.logAction, logActionClass)}>
+                    {action}
+                </div>
+                <div className={s.logActionContent}>
+                    {`${entity} (${objectId})`}
+                </div>
+                <div className={s.logActionButtons}>
+                    <FaUndo />
+                </div>
+            </div>
+        );
+    }
+
+    private getSummaryAction = (logs: ILogEntry[]) => {
+        let action = null;
+        if (
+            logs.some(ev => ev.action === logActions.ADD) &&
+            logs.some(ev => ev.action === logActions.DELETE)
+        ) {
+            action = null;
+        } else if (
+            logs.some(ev => ev.action === logActions.ADD)
+        ) {
+            action = logActions.ADD;
+        } else if (
+            logs.some(ev => ev.action === logActions.DELETE)
+        ) {
+            action = logActions.DELETE;
+        } else {
+            action = logActions.MOVE;
+        }
+        return action;
     }
 
     private getLogs = () => {
-        return 'logs';
+        const groupedById = GeometryLogStore!.log.reduce(
+            (rv, x) => {
+                (rv[x.objectId] = rv[x.objectId] || []).push(x);
+                return rv;
+            },
+            {},
+        );
+        const res : JSX.Element[] = [];
+        Object.entries(groupedById).forEach(([id, groupedLogs]: [string, ILogEntry[]]) => {
+            const first = groupedLogs[0];
+            res.push(this.getLogRow(
+                this.getSummaryAction(groupedLogs),
+                id,
+                first.entity,
+            ));
+        });
+
+        return res;
```

Comment:
This needs rewriting as it seems way too complex considering that we only show ADD events at this point.

Why not just use something like:

```
const events = this.props.GeometryLogStore!.events

return (
   { this.events.forEach.map(event) => {
        return (
              <EventLogRow event={event} />
        )
     }
   }
)
```

Have EventLogRow in its own component.

Reply:
Your suggestion would not take into consideration modifying the same item twice. Your suggestion would in that case print both events, when we have agreed on only showing the entry once.
Having EventLogRow in a separate component is a good advice.

Reply:
I think we are still mixing two concepts, eventLog and showing what changes user has made (changeLog). EventLog !== changeLog. We are implementing eventLog here.

If we have the following case: user adds a link, removes a link, adds the link back again

EventLog shows:
Add link
Remove link
Add link

ChangeLog shows:
Add link

I made a more simplified code that you should use here since we are simply rendering through a list here:

```
    private renderEventRow(action: logActions, entity: entityNames) {
        let eventTypeClass = '';
        switch (action) {
        case logActions.ADD:
            eventTypeClass = s.eventTypeAdd;
            break;
        // Uncomment when we have use case for edit and deletion
        //
        // case logActions.DELETE:
        //     eventTypeClass = s.eventTypeDelete;
        //     break;
        // case logActions.MOVE:
        //     eventTypeClass = s.eventTypeMove;
        //     break;
        }

        return (
            <div className={s.event}>
                <div className={classnames(s.eventType, eventTypeClass)}>
                    {action}
                </div>
                <div className={s.eventContent}>
                    {entity}
                </div>
                <div className={s.eventButtons}>
                    <FaUndo />
                </div>
            </div>
        );
    }

    private renderEventList = () => {
        const events = GeometryEventStore!.events;
        return events.map(event: IEvent) => {
            this.renderEventRow(event.action, event.entity);
        }
    }
```

Reply:
Okay, I did now remove the grouping logic, so we now have a simpler event log.